exporters/prometheus/exporter.go

```diff
@@ -39,25 +39,53 @@ type Exporter struct {
 // collector is used to implement prometheus.Collector.
 type collector struct {
 	metric.Reader
+	bridges []Bridge
 }
 
 // config is added here to allow for options expansion in the future.
-type config struct{}
+type config struct {
+	reader  metric.Reader
+	bridges []Bridge
+}
 
 // Option may be used in the future to apply options to a Prometheus Exporter config.
 type Option interface {
 	apply(config) config
 }
 
+type optionFunc func(config) config
+
+func (fn optionFunc) apply(cfg config) config {
+	return fn(cfg)
+}
+
+// Bridge is a source of metrics other than the OpenTelemetry SDK.
+type Bridge interface {
+	// Collect gathers and returns all metric data from the Bridge.
+	Collect(context.Context) (metricdata.ScopeMetrics, error)
```

Comment:
Note that this provides a ScopeMetrics, rather than a ResourceMetrics.  This ensures we only get the resource from the SDK.